src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const checkSep = (separator === null) ? 'null' : separator;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += '' + checkSep;
+      } else {
+        result += this[i] + checkSep;
+      }
+    }
+
+    return (this[this.length - 1] === undefined)
+      ? result : result + this[this.length - 1];
```

Comment:
multi-line and generic check?
```suggestion
    return (this[this.length - 1])
      ? result + this[this.length - 1]
      : result;
```

Reply:
yeah i changed it, it works well, but what if the last element is 0 or NaN(the tests not checks this), thats gonna be not added at the end too right? Thank you for the suggestions! :)